temporalio/worker/_workflow_instance.py

```diff
@@ -434,7 +434,6 @@ def is_completion(command):
                 command.HasField("complete_workflow_execution")
                 or command.HasField("continue_as_new_workflow_execution")
                 or command.HasField("fail_workflow_execution")
-                or command.HasField("cancel_workflow_execution")
```

Comment:
I don't think this should be removed, I think we should continue to warn on unfinished handlers regardless of reason for unfinished

Reply:
See https://github.com/temporalio/sdk-python/pull/653#discussion_r1781116087